geotrek/trekking/parsers.py

```diff
@@ -208,3 +216,398 @@ class GeotrekPOIParser(GeotrekParser):
+        gpx = self._fetch_gpx_from_url(plan)
+
+        return ApidaeTrekParser._get_geom_from_gpx(gpx)
+
+    def filter_labels(self, src, val):
+        # TODO: unwrap val into typologies and environnements for clarity
+        filtered_val = []
+        for subval in val:
+            if not subval:
+                continue
+            for item in subval:
+                item_type = item['elementReferenceType']
+                if ((item_type == 'TypologiePromoSitra' and item['id'] in TYPOLOGIES_SITRA_IDS_AS_LABELS)
+                        or (item_type == 'Environnement' and item['id'] in ENVIRONNEMENTS_IDS_AS_LABELS)):
+                    filtered_val.append(item['libelleFr'])
+        return self.apply_filter(
+            dst='labels',
+            src=src,
+            val=filtered_val
+        )
+
+    def filter_themes(self, src, val):
+        return self.apply_filter(
+            dst='themes',
+            src=src,
+            val=[item['libelleFr'] for item in val if item['id'] in TYPOLOGIES_SITRA_IDS_AS_THEMES]
+        )
+
+    def filter_description(self, src, val):
+        ouverture, descriptifs, itineraire = val
+        html_description = defaultdict(lambda: '')
+
+        def append_to_html_description(translated_field, transform_func=ApidaeTrekParser._transform_description_to_html):
+            for lang in settings.MODELTRANSLATION_LANGUAGES:
+                try:
+                    html_description[lang] += transform_func(translated_field[f'libelle{lang.capitalize()}'])
+                except KeyError:
+                    pass
+
+        def get_guidebook():
+            if not descriptifs:
+                return None
+            for d in descriptifs:
+                if d['theme']['id'] == GUIDEBOOK_DESCRIPTION_ID:
+                    return d
+            return None
+
+        guidebook = get_guidebook()
+        if guidebook:
+            append_to_html_description(guidebook['description'], transform_func=ApidaeTrekParser._transform_guidebook_to_html)
+
+        if ouverture:
+            append_to_html_description(ouverture['periodeEnClair'])
+
+        if itineraire:
+            append_to_html_description(ApidaeTrekParser._make_marking_description(itineraire))
+
+        return self.apply_filter(
+            dst='description',
+            src=src,
+            val=html_description
+        )
+
+    def filter_source(self, src, val):
+        manager = val
+        sources = self.apply_filter(
+            dst='source',
+            src=src,
+            val=[manager['nom']]
+        )
+        source = sources[0]
+        source.website = manager['siteWeb']
+        source.save()
+        return sources
+
+    def filter_difficulty(self, src, val):
+        types_clientele = val
+        difficulty_level = None
+        for tc in types_clientele:
+            if tc['id'] in TYPES_CLIENTELE_IDS_AS_DIFFICULTY_LEVELS:
+                difficulty_level = tc
```

Comment:
you could return here ?

```
self.apply_filter(
                dst='difficulty',
                src=src,
                val=difficulty_level[f'libelle{settings.MODELTRANSLATION_DEFAULT_LANGUAGE.capitalize()}']
            )
            ```

Reply:
I do not understand. Did you mean it misses an explicit return when `difficulty_level` is None?

Reply:
OK, the proposition is to return from inside the loop as soon as I find the matching difficulty level. Though shorter I don't like to return from inside clauses. ;)